hp.py

```diff
@@ -81,24 +78,19 @@ def show_dimm(self, raw_status):
     # Module #:                     4
     # Status:                       Ok
     #
-        dimm_status = {}
         dimm_list = []
+
         if raw_status is not None:
-            for line in raw_status.split('\n'):
-                new_line = line.split()
-                if new_line is not None:
-                    if "Processor" in new_line and "#:" in new_line:
-                        # new chunk - add the previous one to the list and create a new dict
-                        dimm_list.append(dimm_status)
-                        dimm_status = {}
-                        dimm_status["Processor"] = new_line[2]
-                    if "Module" in new_line and "#:" in new_line:
-                        dimm_status["Module"] = new_line[2]
-                    if "Status:" in new_line:
-                        dimm_status["Status"] = new_line[1].lower()
-            # add the last dict
-            dimm_list.append(dimm_status)
-        return(dimm_list)
+            for line in raw_status.split("\n"):
+                if "Processor #:" in line:
+                    dimm_status = { "Processor": line.split()[-1] }
+                elif "Module #:" in line:
+                    dimm_status["Module"] = line.split()[-1]
+                elif "Status:" in line:
+                    dimm_status["Status"] = line.split()[-1].lower()
+                    dimm_list.append(dimm_status)
```

Comment:
If Status is always the last line, you can just append to the list when you see it.